### PR TITLE
Add a summary explaining the usage and context for JAX PRNG design.

### LINKED ISSUE #2087: Improve the documentation for jax/random.py with a standard usage
It would be nice to have at the top of the module a few lines example on how this is expected to be used.

it's great that we have the design and history:
"""
The JAX PRNG system is based on "Parallel random numbers: as easy as 1, 2, 3"
(Salmon et al. 2011). For details on the design and its motivation, see:

https://github.com/google/jax/blob/master/design_notes/prng.md
"""
However:
1) we can probably get something simpler than having to read 2 papers + https://github.com/google/jax/blob/master/design_notes/prng.md
2) The current example at the bottom uses "rng = lax.rng.new_rng()" which does not exist
3) it would be nice for readers to get the 5 lines examples explaining how to use it.

Explaining how to use  random.PRNGKey and random.split would be useful. I have not myself understood yet how to use that.

Is the standard way to do the following? if yes, just saying so would help:

key = jax.random.PRNGKey(<seed>)

for _ in range(100):
  new_key, key = jax.random.split(key)
  random_obj = jax.randint(new_key, ...)

### PR DESCRIPTION
**Pull request by jblespiau**
The current design_notes do not match current JAX API, and it's a pretty
long doc to read to understand how to use it.

Closes: #2087

### CODE REVIEW

**Comment by tomhennigan** [comment]
@hawkinsp mind taking a look?

File: jax/random.py [hunk 1]
```diff
@@ -14,10 +14,30 @@
...
 
-The JAX PRNG system is based on "Parallel random numbers: as easy as 1, 2, 3"
-(Salmon et al. 2011). For details on the design and its motivation, see:
-
-https://github.com/google/jax/blob/master/design_notes/prng.md
+Usage:
+
+```python
+rng = jax.random.PRNGKey(seed)
+for i in xrange(num_steps):
```
Comment (hawkinsp) [inline jax/random.py:21]: s/xrange/range/

(Python 2 is gone!)

File: jax/random.py [hunk 2]
```diff
@@ -14,10 +14,30 @@
 
 """JAX pseudo-random number generators (PRNGs).
 
-The JAX PRNG system is based on "Parallel random numbers: as easy as 1, 2, 3"
-(Salmon et al. 2011). For details on the design and its motivation, see:
-
-https://github.com/google/jax/blob/master/design_notes/prng.md
+Usage:
```
Comment (hawkinsp) [inline jax/random.py:17]: Example usage:

File: jax/random.py [hunk 3]
```diff
@@ -14,10 +14,30 @@
...
+```python
+rng = jax.random.PRNGKey(seed)
+for i in xrange(num_steps):
+  rng, rng_input = jax.random.split(rng)
+  params = compiled_update(rng_input, params, next(batches))
+```
+
+Context:
+
+Among other requirements, Jax PRNG aims at:
```
Comment (hawkinsp) [inline jax/random.py:28]: Among other requirements, the JAX PRNG aims to:

File: jax/random.py [hunk 4]
```diff
@@ -14,10 +14,30 @@
...
+and multi-replica, multi-core computation. In particular it should not use 
+sequencing constraints between random function calls.
+
+The approach is based on:
+1. "Parallel random numbers: as easy as 1, 2, 3" (Salmon et al. 2011)
+2. "Splittable pseudorandom number generators using cryptographic hashing"
+(Claessen et al. 2013)
+
+See also https://github.com/google/jax/blob/master/design_notes/prng.md
+for the historical design and its motivation.
```
Comment (hawkinsp) [inline jax/random.py:40]: Remove "historical" ?

Is it out of date in some way? If so, we should update it.

**Review by hawkinsp — CHANGES_REQUESTED** [review]

File: jax/random.py (hunk 1 above)
Comment (jblespiau) [reply jax/random.py:21]: Done.

File: jax/random.py (hunk 2 above)
Comment (jblespiau) [reply jax/random.py:17]: Done.

File: jax/random.py (hunk 3 above)
Comment (jblespiau) [reply jax/random.py:28]: Done.

File: jax/random.py (hunk 4 above)
Comment (jblespiau) [reply jax/random.py:40]: I think the design no longer matches what has been implemented, e.g.

```
rng = lax.rng.new_rng()
for i in xrange(num_steps):
  rng, rng_input = lax.rng.split(rng)
```

So it's kind of explaining the design at some point in time, but does not match exactly what has been done.

**Review by gnecula — APPROVED** [review]
Thanks, sorry for the delay!
